### PR TITLE
Improve OSX install experience

### PR DESCRIPTION
**Pull request by larroy**
Added a semiautomated install script for MacOSX

### CODE REVIEW

File: docs/get_started/osx_setup.md [hunk 1]
```diff
@@ -20,6 +18,33 @@ Install the dependencies, required for MXNet, with the following commands:
 	# Insert the Homebrew directory at the top of your PATH environment variable
 	export PATH=/usr/local/bin:/usr/local/sbin:$PATH
 ```
+
+Initialize the git submodules and fetch them:
```
Comment (piiswrong) [inline docs/get_started/osx_setup.md]: clone with --recursive directly

File: setup-utils/install-mxnet-macosx.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
```
Comment (howard0su) [inline setup-utils/install-mxnet-macosx.sh]: suggest to rename this to install mxnet-osx.sh as normally we refer the os as osx

File: setup-utils/install-mxnet-macosx.sh [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  exit 1
+fi
+
+function brew_install() {
+  echo "Checking for brew formula $1"
+  brew ls --versions $1 || brew install $1
+}
+# You need to have brew installed, check: http://mxnet.io/get_started/osx_setup.html
+#brew update
+brew_install cmake
```
Comment (howard0su) [inline setup-utils/install-mxnet-macosx.sh]: why not use one single brew to install multi packages to speed up the installation?

File: setup-utils/install-mxnet-macosx.sh [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+brew_install opencv
+# For getting pip
+brew_install python
+# For visualization of network graphs
+pip install graphviz
+# Jupyter notebook
+pip install jupyter
+
+
+cp make/osx.mk ./config.mk
```
Comment (howard0su) [inline setup-utils/install-mxnet-macosx.sh]: backup config.mk if necessary? or even skip cp if there is already a config.mk, instead just show a message says config.mk is there.

File: setup-utils/install-mxnet-macosx.sh [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+echo "USE_BLAS = openblas" >> ./config.mk
+echo "ADD_CFLAGS += -I/usr/local/opt/openblas/include" >> ./config.mk
+echo "ADD_LDFLAGS += -L/usr/local/opt/openblas/lib" >> ./config.mk
+echo "ADD_LDFLAGS += -L/usr/local/lib/graphviz/" >> ./config.mk
+make -j$(sysctl -n hw.ncpu)
```
Comment (howard0su) [inline setup-utils/install-mxnet-macosx.sh]: why not just install python as next steps?
cd python ; sudo python setup.py install

Then you need rename your script as others under setup-utils.

File: setup-utils/install-mxnet-macosx.sh (hunk 3 above)
Comment (larroy) [inline setup-utils/install-mxnet-macosx.sh]: Some packages fail to install if you already have it, that's why I used the function to check if it's already installed. How would that work with multiple packages?

File: setup-utils/install-mxnet-macosx.sh (hunk 2 above)
Comment (larroy) [inline setup-utils/install-mxnet-macosx.sh]: Done

File: docs/get_started/osx_setup.md (hunk 1 above)
Comment (larroy) [inline docs/get_started/osx_setup.md]: it's not going to do any harm if the submodules are already there.

**Comment by larroy** [comment]
Thanks for the review. I addressed most of the issues and rebased the changes. Anything else that I should address?

File: setup-utils/install-mxnet-osx.sh [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+echo "USE_BLAS = openblas" >> ./config.mk
+echo "ADD_CFLAGS += -I/usr/local/opt/openblas/include" >> ./config.mk
+echo "ADD_LDFLAGS += -L/usr/local/opt/openblas/lib" >> ./config.mk
+echo "ADD_LDFLAGS += -L/usr/local/lib/graphviz/" >> ./config.mk
+make -j$(sysctl -n hw.ncpu)
+
+echo "Installing the Python mxnet modules..."
+
+cd python
+sudo python setup.py install
```
Comment (piiswrong) [inline setup-utils/install-mxnet-osx.sh]: we usually recommend installing inplace. i.e. clone the repo to ~/mxnet and add ~/mxnet/python to $PYTHONPATH. It's easier to use the exmaples and modify code that way.

**Comment by larroy** [comment]
Thanks for the merge!  👍

**Pull request merged**